source/extensions/filters/network/postgres_proxy/postgres_filter.cc

```diff
@@ -159,6 +164,29 @@ void PostgresFilter::incStatements(StatementType type) {
   }
 }
 
+void PostgresFilter::processQuery(const std::string& sql) {
```

Comment:
naive q: are you buffering the query payload until you get the entire query? I am guessing you are.

Reply:
and I also hope you ahve logic like
```
if (sniffing_) {
    write_buffer_.add(data);
    doDecode(write_buffer_);
  }
  return Network::FilterStatus::Continue;
```

like the way we have in mongo and mysql?

Reply:
Yes, we are buffering until we get the entire message. The wire protocol between postgres and mysql is obviously different. For postgres, there is a length field in the header which we extract and buffer until we get the entire message. Then we parse it. The only exception from this rule is when at the startup handshake we detect that session is going to be encrypted and there is no buffering then. Whatever is received, it is passed to the next filter.